sample_factory/algo/sampling/non_batched_sampling.py

```diff
@@ -121,7 +121,8 @@ def set_trajectory_data(self, data: Dict, rollout_step: int):
         self.curr_traj_buffer[rollout_step] = data
 
     def reset_rnn_state(self):
-        self.last_rnn_state[:] = 0.0
+        if self.cfg.use_rnn:
```

Comment:
Does this serve any specific purpose, or just for clarity?
If use_rnn is False, then we initialize rnn_state tensors to be size 1 and self.last_rnn_state[:] = 0.0 is a cheap operation.
Anyways, I don't mind this change, but if this was to fix some bug please let me know or add a comment! :)

Reply:
When I try use_rnn = False in non-batched, I get an error of "IndexError: too many indices for array" when doing "self.last_rnn_state[:] = 0.0"

Reply:
Okay, gotcha. Good find!
I believe the problem is elsewhere, we should not be fixing this line. Here: https://github.com/alex-petrenko/sample-factory/commit/4275daf2ef1cf49e68617b40dc9f2a432abbe118

I think this fixes it. I don't know why that squeeze was needed there. I tried on a couple of envs and it seems to work. Pull my changes.